crates/hulk_nao/src/microphones.rs

```diff
@@ -17,32 +21,28 @@ pub struct Microphones {
 
 impl Microphones {
     pub fn new(parameters: Parameters) -> Result<Self> {
-        let device = PCM::new("default", Direction::Capture, false)
-            .wrap_err("failed to open audio device")?;
-        {
-            let hardware_parameters =
-                HwParams::any(&device).wrap_err("failed to create hardware parameters")?;
-            hardware_parameters
-                .set_access(parameters.access)
-                .wrap_err("failed to set access")?;
-            hardware_parameters
-                .set_format(parameters.format)
-                .wrap_err("failed to set format")?;
-            hardware_parameters
-                .set_rate_near(parameters.sample_rate, ValueOr::Nearest)
-                .wrap_err("failed to set sample rate")?;
-            hardware_parameters
-                .set_channels(parameters.number_of_channels as u32)
-                .wrap_err("failed to set channel")?;
-            device
-                .hw_params(&hardware_parameters)
-                .wrap_err("failed to set hardware parameters")?;
-        }
-        device.prepare().wrap_err("failed to prepare device")?;
+        let device = create_device(&parameters).wrap_err("failed to create device")?;
         Ok(Self { device, parameters })
     }
 
-    pub fn read_from_microphones(&self) -> Result<Samples> {
+    pub fn retrying_read(&mut self) -> Result<Samples> {
+        let number_of_retries = 5;
```

Comment:
You defined this in hardware parameters but are not using it.

Reply:
oooooops :sweat_smile: :rocket: 

Fixed